<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Open peer review: reproducibility | Portfolio</title>
  <meta name="description" content="8 Open peer review: reproducibility | Portfolio" />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Open peer review: reproducibility | Portfolio" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Open peer review: reproducibility | Portfolio" />
  
  
  

<meta name="author" content="Cheryl" />


<meta name="date" content="2025-01-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="c.elegans-plate-experiment.html"/>
<link rel="next" href="create-a-package.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="libs/typedarray-0.1/typedarray.min.js"></script>
<link href="libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="libs/plotly-main-2.11.1/plotly-latest.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introductie</a></li>
<li class="chapter" data-level="2" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html"><i class="fa fa-check"></i><b>2</b> Curriculum vitae</a>
<ul>
<li class="chapter" data-level="2.1" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#opleidingen"><i class="fa fa-check"></i><b>2.1</b> Opleidingen</a></li>
<li class="chapter" data-level="2.2" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#werkervaring"><i class="fa fa-check"></i><b>2.2</b> Werkervaring</a></li>
<li class="chapter" data-level="2.3" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#vaardigheden"><i class="fa fa-check"></i><b>2.3</b> Vaardigheden</a></li>
<li class="chapter" data-level="2.4" data-path="curriculum-vitae.html"><a href="curriculum-vitae.html#cursussen"><i class="fa fa-check"></i><b>2.4</b> Cursussen</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="looking-ahead.html"><a href="looking-ahead.html"><i class="fa fa-check"></i><b>3</b> Looking ahead</a></li>
<li class="chapter" data-level="4" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html"><i class="fa fa-check"></i><b>4</b> R for Mass Spectrometry</a>
<ul>
<li class="chapter" data-level="4.1" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#introductie-1"><i class="fa fa-check"></i><b>4.1</b> Introductie</a></li>
<li class="chapter" data-level="4.2" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#planning"><i class="fa fa-check"></i><b>4.2</b> Planning</a></li>
<li class="chapter" data-level="4.3" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#uitvoering-met-voorbeeld-data-pxdooooo1"><i class="fa fa-check"></i><b>4.3</b> Uitvoering met voorbeeld data PXDOOOOO1</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#inlezen-van-de-data-in-r"><i class="fa fa-check"></i><b>4.3.1</b> Inlezen van de data in R</a></li>
<li class="chapter" data-level="4.3.2" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#ruwe-data-visualisatie-verwerking-en-manipulatie"><i class="fa fa-check"></i><b>4.3.2</b> Ruwe data visualisatie, verwerking en manipulatie</a></li>
<li class="chapter" data-level="4.3.3" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#identification-data"><i class="fa fa-check"></i><b>4.3.3</b> Identification data</a></li>
<li class="chapter" data-level="4.3.4" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#msnbase-package"><i class="fa fa-check"></i><b>4.3.4</b> MSnbase package</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#uitvoering-met-ms-data-pxd008990"><i class="fa fa-check"></i><b>4.4</b> Uitvoering met MS data PXD008990</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#inlezen-van-de-data-in-r-1"><i class="fa fa-check"></i><b>4.4.1</b> Inlezen van de data in R</a></li>
<li class="chapter" data-level="4.4.2" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#ruwe-data-visualisatie-verwerking-en-manipulatie-1"><i class="fa fa-check"></i><b>4.4.2</b> Ruwe data visualisatie, verwerking en manipulatie</a></li>
<li class="chapter" data-level="4.4.3" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#samenvattings-opdracht"><i class="fa fa-check"></i><b>4.4.3</b> Samenvattings opdracht</a></li>
<li class="chapter" data-level="4.4.4" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#msnbase-package-1"><i class="fa fa-check"></i><b>4.4.4</b> MSnbase package</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#uitvoering-met-ms-data-pxd018226"><i class="fa fa-check"></i><b>4.5</b> Uitvoering met MS data PXD018226</a>
<ul>
<li class="chapter" data-level="4.5.1" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#inlezen-van-data-in-r"><i class="fa fa-check"></i><b>4.5.1</b> Inlezen van data in R</a></li>
<li class="chapter" data-level="4.5.2" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#ruwe-data-visualisatie"><i class="fa fa-check"></i><b>4.5.2</b> Ruwe data visualisatie</a></li>
<li class="chapter" data-level="4.5.3" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#identification-data-1"><i class="fa fa-check"></i><b>4.5.3</b> Identification data</a></li>
<li class="chapter" data-level="4.5.4" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#combineren-van-identificatie-data-met-ruwe-data"><i class="fa fa-check"></i><b>4.5.4</b> Combineren van identificatie data met ruwe data</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="r-for-mass-spectrometry.html"><a href="r-for-mass-spectrometry.html#links"><i class="fa fa-check"></i><b>4.6</b> Links</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html"><i class="fa fa-check"></i><b>5</b> R Seurat: Guided Clustering Tutorial</a>
<ul>
<li class="chapter" data-level="5.1" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#uitvoering-met-voorbeeld-data"><i class="fa fa-check"></i><b>5.1</b> Uitvoering met voorbeeld data</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#pre-processing-workflow"><i class="fa fa-check"></i><b>5.1.1</b> Pre-processing workflow</a></li>
<li class="chapter" data-level="5.1.2" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#identification-of-highly-variable-features"><i class="fa fa-check"></i><b>5.1.2</b> Identification of highly variable features</a></li>
<li class="chapter" data-level="5.1.3" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#perform-linear-dimensional-reduction"><i class="fa fa-check"></i><b>5.1.3</b> Perform linear dimensional reduction</a></li>
<li class="chapter" data-level="5.1.4" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#cluster-the-cells-umap"><i class="fa fa-check"></i><b>5.1.4</b> Cluster the cells &amp; UMAP</a></li>
<li class="chapter" data-level="5.1.5" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#finding-differentially-expressed-features"><i class="fa fa-check"></i><b>5.1.5</b> Finding differentially expressed features</a></li>
<li class="chapter" data-level="5.1.6" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#assigning-cell-type-identity-to-clusters"><i class="fa fa-check"></i><b>5.1.6</b> Assigning cell type identity to clusters</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#uitvoering-met-csf1r-related-disorder"><i class="fa fa-check"></i><b>5.2</b> Uitvoering met CSF1R-related disorder</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#pre-processing-workflow-1"><i class="fa fa-check"></i><b>5.2.1</b> Pre-processing workflow</a></li>
<li class="chapter" data-level="5.2.2" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#identification-of-highly-variable-features-1"><i class="fa fa-check"></i><b>5.2.2</b> Identification of highly variable features</a></li>
<li class="chapter" data-level="5.2.3" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#perform-linear-dimensional-reduction-1"><i class="fa fa-check"></i><b>5.2.3</b> Perform linear dimensional reduction</a></li>
<li class="chapter" data-level="5.2.4" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#cluster-the-cells-umap-1"><i class="fa fa-check"></i><b>5.2.4</b> Cluster the cells &amp; UMAP</a></li>
<li class="chapter" data-level="5.2.5" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#finding-differentially-expressed-features-1"><i class="fa fa-check"></i><b>5.2.5</b> Finding differentially expressed features</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="r-seurat-guided-clustering-tutorial.html"><a href="r-seurat-guided-clustering-tutorial.html#links-1"><i class="fa fa-check"></i><b>5.3</b> Links</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="guerrilla-analytics-framework.html"><a href="guerrilla-analytics-framework.html"><i class="fa fa-check"></i><b>6</b> Guerrilla Analytics framework</a>
<ul>
<li class="chapter" data-level="6.1" data-path="guerrilla-analytics-framework.html"><a href="guerrilla-analytics-framework.html#daur2"><i class="fa fa-check"></i><b>6.1</b> Daur2</a></li>
<li class="chapter" data-level="6.2" data-path="guerrilla-analytics-framework.html"><a href="guerrilla-analytics-framework.html#workflows"><i class="fa fa-check"></i><b>6.2</b> Workflows</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="c.elegans-plate-experiment.html"><a href="c.elegans-plate-experiment.html"><i class="fa fa-check"></i><b>7</b> <em>C.elegans</em> plate experiment</a></li>
<li class="chapter" data-level="8" data-path="open-peer-review-reproducibility.html"><a href="open-peer-review-reproducibility.html"><i class="fa fa-check"></i><b>8</b> Open peer review: reproducibility</a>
<ul>
<li class="chapter" data-level="8.1" data-path="open-peer-review-reproducibility.html"><a href="open-peer-review-reproducibility.html#part-1"><i class="fa fa-check"></i><b>8.1</b> Part 1</a></li>
<li class="chapter" data-level="8.2" data-path="open-peer-review-reproducibility.html"><a href="open-peer-review-reproducibility.html#part-2"><i class="fa fa-check"></i><b>8.2</b> Part 2</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="create-a-package.html"><a href="create-a-package.html"><i class="fa fa-check"></i><b>9</b> Create a package</a></li>
<li class="chapter" data-level="10" data-path="projecticum-antimicrobiële-resistentie-amr.html"><a href="projecticum-antimicrobiële-resistentie-amr.html"><i class="fa fa-check"></i><b>10</b> Projecticum Antimicrobiële Resistentie (AMR)</a></li>
<li class="chapter" data-level="11" data-path="ecdc-covid-19-cases-data-parameterization.html"><a href="ecdc-covid-19-cases-data-parameterization.html"><i class="fa fa-check"></i><b>11</b> ECDC COVID-19 cases data parameterization</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Portfolio</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="open-peer-review-reproducibility" class="section level1 hasAnchor" number="8">
<h1><span class="header-section-number">8</span> Open peer review: reproducibility<a href="open-peer-review-reproducibility.html#open-peer-review-reproducibility" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="part-1" class="section level2 hasAnchor" number="8.1">
<h2><span class="header-section-number">8.1</span> Part 1<a href="open-peer-review-reproducibility.html#part-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Open science stelt onderzoeken meer open voor participatie, beoordeling/weerlegging, verbetering en hergebruik. Repita heeft acht transparantie criteria opgesteld waarmee de reproduceerbaarheid van een onderzoek geëvalueerd kan worden.
Het artikel dat in deze opdracht wordt geëvalueerd is “Glycomic and Proteomic Changes in Aging Brain Nigrostriatal Pathway”. Het algemene doel van dit artikel is het bepalen van het effect van leeftijdsgerelateerde veranderingen in glycaan receptoren op de binding van Adeno-associated virussen. Gen therapie met Adeno-associated virussen is succesvol geweest in pre-klinische modellen van de ziekte van Parkinson, echter bij klinische trials met mensen is het niet efficiënt bewezen.</p>
<p>Het leeftijdsonderzoek werd uitgevoerd op zes jonge en zes oude mannelijke F344 ratten. De hersenen van de ratten zijn post-mortum gesectioneerd, na meerdere rondes van digestie konden hieruit de te onderzoeken glycan en proteomen worden geëxtraheert. Vervolgens is er een HILIC-MS of LC-MS analyse uitgevoerd op de samples, waarna de data is geanalyseerd met verscheidene pakketten en softwares.</p>
<p>Verschillende brein regio-specifieke HS disaccharide profielen waren waargenomen in oude vergeleken met jonge ratten. Ook waren er brein regio- en leeftijdsspecifieke N-glycaan profielen waargenomen. In het nigrostriatale pathway van oude ratten zijn ook veranderingen in brein regio-specifieke eiwit-expressie waargenomen.</p>
<p>De evaluatie van de acht criteria is in het onderstaande tabel weergeven.
Per criteria zal ik nog verdere uitleg geven waarom ik voor de score heb gekozen.
De aanleiding voor het artikel is aanwezig, de onderzoekers vertellen eerst over AAV als gen therapie en hoe dit niet effectief is in mensen. Hierbij geven ze aan waarom zij denken dat het niet werkt en gaan vervolgens in het onderzoek achterhalen of dit klopt.
Er is een data availability statement aanwezig, de data is bereikbaar op de PRIDE repository met project accession nummer PXD008990.
De locatie van de data is ook weergeven, met een link naar de PRIDE repository; <a href="https://www.ebi.ac.uk/pride/archive?keyword=PXD008990" class="uri">https://www.ebi.ac.uk/pride/archive?keyword=PXD008990</a>.
De study location is niet genoemd in de methodes van het artikel.
Author review heeft als score Tier 3, dit houdt in dat de auteur zijn/haar instituuts e-mail adress heeft gebruikt als primair contact. Wel is dit maar bij een van de auteurs genoteerd.
Een ethics statement is ook niet aanwezig in dit artikel.
Een funding statement is weergeven, de funding was verstrekt door NIH grants P41GM104603, R21CA177479, and R56AG052328.
De code availability is ook weergeven, onder andere Peak Studio versie 8.0 en GlyReSoft versie 0.3.1 waren gebruikt voor de analyse van de data.</p>
<p>6 van de 8 criteria zijn transparant, hiermee zou ik concluderen dat de data reproduceerbaar is, maar er is wel nog ruimte voor verbetering. De ethische statements en de study locatie moeten nog weergeven worden om te voldoen aan de 8 criteria.</p>
<div class="sourceCode" id="cb195"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb195-1"><a href="open-peer-review-reproducibility.html#cb195-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb195-2"><a href="open-peer-review-reproducibility.html#cb195-2" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">kable</span>(</span>
<span id="cb195-3"><a href="open-peer-review-reproducibility.html#cb195-3" tabindex="-1"></a>  <span class="fu">tibble</span>(</span>
<span id="cb195-4"><a href="open-peer-review-reproducibility.html#cb195-4" tabindex="-1"></a>    <span class="st">&quot;Transparency Criteria&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Study purpose&quot;</span>, <span class="st">&quot;Data availability statement&quot;</span>, <span class="st">&quot;Data location&quot;</span>, <span class="st">&quot;Study location&quot;</span>, <span class="st">&quot;Author review&quot;</span>, <span class="st">&quot;Ethics statement&quot;</span>, <span class="st">&quot;Funding statement&quot;</span>, <span class="st">&quot;Code availability&quot;</span>),</span>
<span id="cb195-5"><a href="open-peer-review-reproducibility.html#cb195-5" tabindex="-1"></a>    <span class="st">&quot;Definition&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;A concise statement in the introduction of the article, often in the last paragraph, that establishes the reason the research was conducted. Also called the study objective.&quot;</span>, <span class="st">&quot;A statement, in an individual section offset from the main body of text, that explains how or if one can access a study’s data. The title of the section may vary, but it must explicitly mention data; it is therefore distinct from a supplementary materials section.&quot;</span>, <span class="st">&quot;Where the article’s data can be accessed, either raw or processed.&quot;</span>, <span class="st">&quot;Author has stated in the methods section where the study took place or the data’s country/region of origin.&quot;</span>, <span class="st">&quot;The professionalism of the contact information that the author has provided in the manuscript.&quot;</span>, <span class="st">&quot;A statement within the manuscript indicating any ethical concerns, including the presence of sensitive data.&quot;</span>, <span class="st">&quot;A statement within the manuscript indicating whether or not the authors received funding for their research.&quot;</span>, <span class="st">&quot;Authors have shared access to the most updated code that they used in their study, including code used for analysis.&quot;</span>),</span>
<span id="cb195-6"><a href="open-peer-review-reproducibility.html#cb195-6" tabindex="-1"></a>    <span class="st">&quot;Response Type&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Binary&quot;</span>, <span class="st">&quot;Binary&quot;</span>, <span class="st">&quot;Found Value&quot;</span>, <span class="st">&quot;Binary; Found Value&quot;</span>, <span class="st">&quot;Found Value&quot;</span>, <span class="st">&quot;Binary&quot;</span>, <span class="st">&quot;Binary&quot;</span>, <span class="st">&quot;Binary&quot;</span>),</span>
<span id="cb195-7"><a href="open-peer-review-reproducibility.html#cb195-7" tabindex="-1"></a>    <span class="st">&quot;Score&quot;</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;External Repository; PRIDE&quot;</span>, <span class="st">&quot;No&quot;</span>, <span class="st">&quot;Tier 3&quot;</span>, <span class="st">&quot;No&quot;</span>, <span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;Yes&quot;</span>)</span>
<span id="cb195-8"><a href="open-peer-review-reproducibility.html#cb195-8" tabindex="-1"></a>  ))</span></code></pre></div>
<table>
<colgroup>
<col width="8%" />
<col width="77%" />
<col width="5%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Transparency Criteria</th>
<th align="left">Definition</th>
<th align="left">Response Type</th>
<th align="left">Score</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Study purpose</td>
<td align="left">A concise statement in the introduction of the article, often in the last paragraph, that establishes the reason the research was conducted. Also called the study objective.</td>
<td align="left">Binary</td>
<td align="left">Yes</td>
</tr>
<tr class="even">
<td align="left">Data availability statement</td>
<td align="left">A statement, in an individual section offset from the main body of text, that explains how or if one can access a study’s data. The title of the section may vary, but it must explicitly mention data; it is therefore distinct from a supplementary materials section.</td>
<td align="left">Binary</td>
<td align="left">Yes</td>
</tr>
<tr class="odd">
<td align="left">Data location</td>
<td align="left">Where the article’s data can be accessed, either raw or processed.</td>
<td align="left">Found Value</td>
<td align="left">External Repository; PRIDE</td>
</tr>
<tr class="even">
<td align="left">Study location</td>
<td align="left">Author has stated in the methods section where the study took place or the data’s country/region of origin.</td>
<td align="left">Binary; Found Value</td>
<td align="left">No</td>
</tr>
<tr class="odd">
<td align="left">Author review</td>
<td align="left">The professionalism of the contact information that the author has provided in the manuscript.</td>
<td align="left">Found Value</td>
<td align="left">Tier 3</td>
</tr>
<tr class="even">
<td align="left">Ethics statement</td>
<td align="left">A statement within the manuscript indicating any ethical concerns, including the presence of sensitive data.</td>
<td align="left">Binary</td>
<td align="left">No</td>
</tr>
<tr class="odd">
<td align="left">Funding statement</td>
<td align="left">A statement within the manuscript indicating whether or not the authors received funding for their research.</td>
<td align="left">Binary</td>
<td align="left">Yes</td>
</tr>
<tr class="even">
<td align="left">Code availability</td>
<td align="left">Authors have shared access to the most updated code that they used in their study, including code used for analysis.</td>
<td align="left">Binary</td>
<td align="left">Yes</td>
</tr>
</tbody>
</table>
</div>
<div id="part-2" class="section level2 hasAnchor" number="8.2">
<h2><span class="header-section-number">8.2</span> Part 2<a href="open-peer-review-reproducibility.html#part-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>Geselecteerd project:</em> <br>
Link: <a href="https://osf.io/324jm/" class="uri">https://osf.io/324jm/</a></p>
<p><em>Beschrijf in eigen worden wat de intentie van de code is.</em> <br>
Met behulp van algemene additieve gemengde modellen (GAMMs) proberen ze na te gaan wat het effect van lineaire en log-getransformeerde cloze probability op het verband tussen de woordvoorspelbaarheid en de leestijd.</p>
<p><em>In terms of readibility of the code, how would you grade (1(very bad)-5(very good)) the code available.</em> <br>
Ik zou het een 4 geven, er wordt niet duidelijk uitgelegd waarom bepaalde functies worden gebruikt of wat er wordt uitgevoerd. Verder is de code wel duidelijk en is er goede structuur waarbij headers zijn gebruikt.</p>
<p><em>Run the script or code that is available to reproduce at least 1 figure</em></p>
<p><em>Experiment 1</em> <br></p>
<p>Raw cloze probability:</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="open-peer-review-reproducibility.html#cb196-1" tabindex="-1"></a>dat <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file=</span><span class="st">&quot;../../Downloads/osfstorage-archive/data_and_materials/SPRT_LogLin_216.csv&quot;</span>,<span class="at">head=</span>T)</span>
<span id="cb196-2"><a href="open-peer-review-reproducibility.html#cb196-2" tabindex="-1"></a></span>
<span id="cb196-3"><a href="open-peer-review-reproducibility.html#cb196-3" tabindex="-1"></a>lin_gam<span class="ot">=</span>mgcv<span class="sc">::</span><span class="fu">gam</span>(SUM_3RT_trimmed <span class="sc">~</span><span class="fu">s</span>(c_cloze) <span class="sc">+</span> <span class="fu">s</span>(SUB,<span class="at">bs=</span><span class="st">&quot;re&quot;</span>)<span class="sc">+</span></span>
<span id="cb196-4"><a href="open-peer-review-reproducibility.html#cb196-4" tabindex="-1"></a>              <span class="fu">s</span>(ITEM,<span class="at">bs=</span> <span class="st">&quot;re&quot;</span>),</span>
<span id="cb196-5"><a href="open-peer-review-reproducibility.html#cb196-5" tabindex="-1"></a>             <span class="at">data=</span>dat, <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span>
<span id="cb196-6"><a href="open-peer-review-reproducibility.html#cb196-6" tabindex="-1"></a><span class="co">#summary(lin_gam)</span></span>
<span id="cb196-7"><a href="open-peer-review-reproducibility.html#cb196-7" tabindex="-1"></a>mgcv<span class="sc">::</span><span class="fu">plot.gam</span>(lin_gam, <span class="at">select =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb197"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb197-1"><a href="open-peer-review-reproducibility.html#cb197-1" tabindex="-1"></a><span class="co">#y_pred &lt;- predict(lin_gam, data.frame(x = x_new)) </span></span></code></pre></div>
<p>Log-transformed cloze probability:</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="open-peer-review-reproducibility.html#cb198-1" tabindex="-1"></a>log_gam<span class="ot">=</span>mgcv<span class="sc">::</span><span class="fu">gam</span>(SUM_3RT_trimmed <span class="sc">~</span> <span class="fu">s</span>(c_log_cloze) <span class="sc">+</span> <span class="fu">s</span>(SUB,<span class="at">bs=</span><span class="st">&quot;re&quot;</span>)<span class="sc">+</span></span>
<span id="cb198-2"><a href="open-peer-review-reproducibility.html#cb198-2" tabindex="-1"></a>              <span class="fu">s</span>(ITEM,<span class="at">bs=</span> <span class="st">&quot;re&quot;</span>),</span>
<span id="cb198-3"><a href="open-peer-review-reproducibility.html#cb198-3" tabindex="-1"></a>            <span class="at">data=</span>dat,</span>
<span id="cb198-4"><a href="open-peer-review-reproducibility.html#cb198-4" tabindex="-1"></a>            <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span>
<span id="cb198-5"><a href="open-peer-review-reproducibility.html#cb198-5" tabindex="-1"></a>mgcv<span class="sc">::</span><span class="fu">plot.gam</span>(log_gam, <span class="at">select =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
<p><em>Experiment 2</em> <br></p>
<p>Trying to reproduce GAMMs analysis. Raw cloze probability:</p>
<div class="sourceCode" id="cb199"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb199-1"><a href="open-peer-review-reproducibility.html#cb199-1" tabindex="-1"></a>dat2 <span class="ot">=</span> <span class="fu">read.csv</span>(<span class="at">file=</span><span class="st">&quot;data_and_materials/LogLin_Naming.csv&quot;</span>,<span class="at">head=</span>T)</span>
<span id="cb199-2"><a href="open-peer-review-reproducibility.html#cb199-2" tabindex="-1"></a></span>
<span id="cb199-3"><a href="open-peer-review-reproducibility.html#cb199-3" tabindex="-1"></a>lin_gam<span class="ot">=</span>mgcv<span class="sc">::</span><span class="fu">gam</span>(TRIM_RT <span class="sc">~</span><span class="fu">s</span>(c_cloze) <span class="sc">+</span> <span class="fu">s</span>(subject,<span class="at">bs=</span><span class="st">&quot;re&quot;</span>)<span class="sc">+</span></span>
<span id="cb199-4"><a href="open-peer-review-reproducibility.html#cb199-4" tabindex="-1"></a>              <span class="fu">s</span>(item,<span class="at">bs=</span> <span class="st">&quot;re&quot;</span>),</span>
<span id="cb199-5"><a href="open-peer-review-reproducibility.html#cb199-5" tabindex="-1"></a>            <span class="at">data=</span>dat2, <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span>
<span id="cb199-6"><a href="open-peer-review-reproducibility.html#cb199-6" tabindex="-1"></a>mgcv<span class="sc">::</span><span class="fu">plot.gam</span>(lin_gam, <span class="at">select =</span> <span class="dv">1</span>)</span></code></pre></div>
<p>Log-transformed cloze probability:</p>
<div class="sourceCode" id="cb200"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb200-1"><a href="open-peer-review-reproducibility.html#cb200-1" tabindex="-1"></a>log_gam<span class="ot">=</span>mgcv<span class="sc">::</span><span class="fu">gam</span>(TRIM_RT <span class="sc">~</span> <span class="fu">s</span>(c_log_cloze) <span class="sc">+</span> <span class="fu">s</span>(subject,<span class="at">bs=</span><span class="st">&quot;re&quot;</span>)<span class="sc">+</span></span>
<span id="cb200-2"><a href="open-peer-review-reproducibility.html#cb200-2" tabindex="-1"></a>              <span class="fu">s</span>(item,<span class="at">bs=</span> <span class="st">&quot;re&quot;</span>),</span>
<span id="cb200-3"><a href="open-peer-review-reproducibility.html#cb200-3" tabindex="-1"></a>            <span class="at">data=</span>dat2,</span>
<span id="cb200-4"><a href="open-peer-review-reproducibility.html#cb200-4" tabindex="-1"></a>            <span class="at">method =</span> <span class="st">&quot;REML&quot;</span>)</span>
<span id="cb200-5"><a href="open-peer-review-reproducibility.html#cb200-5" tabindex="-1"></a>mgcv<span class="sc">::</span><span class="fu">plot.gam</span>(log_gam, <span class="at">select =</span> <span class="dv">1</span>)</span></code></pre></div>
<p><em>Taken together on a scale from 1 (very hard) to 5 (very easy), how much effort did it take you to reproduce the visualization from the project, report or article?</em> <br>
Ik zou het overall een 3,5 geven, ik heb de visualisatie van 2 experimenten uitgevoerd. Experiment 1 ging in een keer zonder errors of flaws. Bij experiment 2 kreeg ik wel een error: ‘names’ attribute [1] must be the same length as the vector [0]. Aangezien ik wat data/ uitleg bij deze code mis lukt het me niet over welke attribute en vector het gaat. Het is me niet gelukt om deze code op te lossen.</p>
<p>Ik mis verder wel metadata en een README file.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="c.elegans-plate-experiment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="create-a-package.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
